auraed/src/ebpf/perf_buffer_reader.rs

```diff
@@ -32,11 +32,19 @@ const PER_CPU_BUFFER_SIZE_IN_PAGES: usize = 2;
 
 pub trait PerfBufferReader<T: Clone + Send + 'static> {
     fn read_from_perf_buffer(
-        bpf: &mut Bpf,
+        bpf: &mut Ebpf,
         perf_buffer: &'static str,
     ) -> anyhow::Result<PerfEventBroadcast<T>> {
         // Query the number of CPUs on the host
-        let num_cpus = nr_cpus()?;
+        let num_cpus = nr_cpus().map_err(|(path, error)| {
+            std::io::Error::new(
+                error.kind(),
+                format!(
+                    "Failed to get the number of CPUs at {}: {}",
+                    path, error
+                ),
+            )
+        })?;
```

Comment:
I'm sure there is a more elegant solution.
`nr_cpus()` now returns `Result<size, (&'static str, io::Error)>`.
A quick fix would be `.map_err(|(_, error)| error)` but we'd lose the path. Idk if the path is important information here.

Reply:
> I'm sure there is a more elegant solution.
> `nr_cpus()` now returns `Result<size, (&'static str, io::Error)>`.
> A quick fix would be `.map_err(|(_, error)| error)` but we'd lose the path. Idk if the path is important information here.

I can't think of one either. an odd decision by Aya where an io::Error could be constructed with the path in, or a custom Error could be used...  but while I don't think the path is that important this approach is fine.